src/main/java/games/strategy/triplea/TripleAUnit.java

```diff
@@ -401,14 +400,15 @@ public static int getProductionPotentialOfTerritory(final Collection<Unit> units
 
   public static Unit getBiggestProducer(final Collection<Unit> units, final Territory producer, final PlayerID player,
       final GameData data, final boolean accountForDamage) {
-    final CompositeMatchAnd<Unit> factoryMatch = new CompositeMatchAnd<>(
-        Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player), Matches.unitIsBeingTransported().invert());
+    final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.<Unit>newCompositeBuilder()
```

Comment:
Sorry, should have pointed this out when I submitted the PR...  The answer is unfortunately, no; not as coded.

If the call to `newCompositeBuilder()` was the entire RHS expression, then the compiler would be able to infer the generic type argument based on the LHS type.

However, once I call a method (`add()`) on the RHS value _before the assignment_, the compiler needs to know the generic type at that point.  Since it can't wait for the inference during the assignment to kick in, it needs some help in the form of the explicit type argument.

The alternative is to keep the assignment "clean," and chain the `add()` calls using the bound variable on the next line:

```java
final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.<Unit>newCompositeBuilder();
factoryMatchBuilder.add(Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player))
    .add(Matches.unitIsBeingTransported().invert());
```

Or to keep the `add()` calls lined up like we typically do with `Stream`s:

```java
final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.newCompositeBuilder();
factoryMatchBuilder
    .add(Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player))
    .add(Matches.unitIsBeingTransported().invert());
```

There's some other alternatives:

* Add the overload `Match#newCompositeBuilder(Match<T>...)`.  That will allow the compiler to infer the generic type based on its arguments (if one or more arguments are present) or based on the assignment (if no arguments are present).  For example:

```java
final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.newCompositeBuilder(
    Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player),
    Matches.unitIsBeingTransported().invert());
```

* Add the overload `Match#newCompositeBuilder(Class<T>)`.  This provides the type explicitly, but I would argue that the current solution is actually better than this.  For example:

```java
final Match.CompositeBuilder<Unit> factoryMatchBuilder = Match.newCompositeBuilder(Unit.class)
    .add(Matches.unitIsOwnedAndIsFactoryOrCanProduceUnits(player)),
    .add(Matches.unitIsBeingTransported().invert());
```